packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -854,7 +860,10 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       timeout: 300,
     });
 
-    this.oAuthCustomResource.node.addDependency(this.hostedUICustomResourceInputs!.node!.defaultChild!);
+    if (this.hostedUICustomResourceInputs) {
+      this.oAuthCustomResource.node.addDependency(this.hostedUICustomResourceInputs!.node!.defaultChild!);
```

Comment:
if you added `if` can you drop `!` ?

Reply:
if that works you can also lower the eslint threshold . removing `!` should get rid of a warning.